includes/GFGAET_Piwik_HTTP_API.php

```diff
@@ -0,0 +1,154 @@
+<?php
+class GFGAET_Piwik_HTTP_API {
+
+	private $endpoint = ''; // Tracking HTTP API Endpoint
+	private $_id = ''; // Unique Visitor ID
+	private $idsite = ''; // Site ID
+	private $rec = 1; // Required for tracking
+	private $apiv = 1; // API Version (currently v1)
+	private $rand = ''; // Random value to help avoid the tracking request being cached by the browser or a proxy
+	private $action_name = ''; // Event category
+	private $e_c = ''; // Event category
+	private $e_a = ''; // Event action
+	private $e_n = ''; // Event Label
+	private $e_v = ''; // Event Value
+	private $url = ''; // Full URL
+
+	public function init() {
+		$this->_id = $this->create_client_id();
+
+		$ga_options = get_option( 'gravityformsaddon_GFGAET_UA_settings', false );
+		if ( ! isset( $ga_options[ 'gravity_forms_event_tracking_piwik_url' ] ) ) {
+			$this->endpoint = 'http://www.example.com/piwik/piwik.php'; // Default Tracking HTTP API Endpoint
```

Comment:
The PR looks good, but I'm concerned about this section and if data is always assumed to be sent to this endpoint. I think a check is needed and if people don't fill out the Piwik data, then there should be a check that ignores sending.

Reply:
Ah, I did take a shortcut there which threw away the data rather than checking if the required endpoint and site id are provided. I also have some spots where I have it still checking if the Google info was provided rather than the Piwik info was provided in some spots.

I'll see about getting that updated. Otherwise, feel free to edit my PR as you see fit!